snapcraft/internal/repo.py

```diff
@@ -299,7 +299,10 @@ def _get(self, apt_cache, package_names):
                 logger.debug(
                     'Marking {!r} (and its dependencies) to be fetched'.format(
                         name))
-                apt_cache[name].mark_install()
+                version = None
+                name_arch, version = _get_pkg_name_parts(name)
+                _set_candidate(apt_cache[name_arch], version)
```

Comment:
The complexity is for `_get()` not `_set_candidate()`.  Sergio brings up a valid point I'll see if there's another way to reduce the complexity.